### PR TITLE
WRQ-19328: Fix React 18.3 warnings in enact

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
In React 19, defaultProps in function components will be removed.
In React 18.3, There are warnings for key spreading

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Remove defaultProps for function components and replaced them with ES6 default parameters.
Pass key directly to JSX without using spread

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-19328

### Comments
Enact-DCO-1.0-Signed-off-by: Juwon Jeong (juwon.jeong@lge.com)

### CODE REVIEW

File: packages/ui/Scroller/Scroller.js [hunk 1]
```diff
@@ -32,9 +32,29 @@ const nop = () => {};
  * @ui
  * @public
  */
+const scrollerDefaultProps = {
```
Comment (MikyungKim) [inline packages/ui/Scroller/Scroller.js:35]: How about moving this definition before the jsdoc?

File: packages/ui/Toggleable/Toggleable.js [hunk 2]
```diff
@@ -152,13 +152,14 @@ const ToggleableHOC = hoc(defaultConfig, (config, Wrapped) => {
...
-	function Toggleable (props) {
+	function Toggleable ({disabled = false, ...rest}) {
+		const props = {disabled, ...rest};
 		const updated = {...props};
 		const propSelected = props[prop];
 
 		const hook = useToggle({
 			defaultSelected: props[defaultPropKey],
-			disabled: props.disabled,
+			disabled: disabled,
```
Comment (MikyungKim) [inline packages/ui/Toggleable/Toggleable.js:162]:
```suggestion
			disabled,
```

File: packages/ui/VirtualList/VirtualList.js [hunk 3]
```diff
@@ -30,9 +30,30 @@ const nop = () => {};
  * @ui
  * @public
  */
+const virtualListDefaultProps = {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/VirtualList.js:33]: Same here as Scroller.

File: packages/ui/VirtualList/VirtualList.js [hunk 4]
```diff
@@ -276,7 +297,18 @@ VirtualList.propTypes = /** @lends ui/VirtualList.VirtualList.prototype */ {
 	verticalScrollbar: PropTypes.oneOf(['auto', 'visible', 'hidden'])
 };
 
-VirtualList.defaultProps = {
+VirtualList.defaultPropValues = virtualListDefaultProps;
+
+/**
```
Comment (MikyungKim) [inline packages/ui/VirtualList/VirtualList.js:302]: Please don't move this jsdoc.

File: packages/ui/Scroller/Scroller.js (hunk 1 above)
Comment (juwonjeong) [reply packages/ui/Scroller/Scroller.js:21]: It's moved

File: packages/ui/VirtualList/VirtualList.js (hunk 3 above)
Comment (juwonjeong) [reply packages/ui/VirtualList/VirtualList.js:24]: It's moved

File: packages/ui/VirtualList/VirtualList.js (hunk 4 above)
Comment (juwonjeong) [reply packages/ui/VirtualList/VirtualList.js]: I didn't move the jsdocs code in anywhere, however git diff shown as if it is moved. I fixed to only add definition before jsdoc.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**